### ISSUE TITLE
Silent sign-out does not invalidate session cookies

### ISSUE THREAD

**Issue by esune** [perarnborg/vuex-oidc#103]
I am trying to get my webapp to silently sign-out without going through the IdP (Keycloak) by calling `unsetOidcAuth`.

This results in the app state being reset correctly, however it appears that the session cookies created by the IdP at login time are not being removed, causing an "auto-signin" event (no authentication page from the IdP is displayed) when accessing the application a second time.

The session cookies are not available in the context of the application, so I am unable to reset them programmatically after calling `unsetOidcAuth`, and performing a sign-out using `signOutOidc` triggers a redirect to the application landing page (as configured in the OIDC settings).

Is it possible to completely destroy the authentication session on the client, without going through the IdP?
If not, could `signOutOidc` accept a `redirect_uri` so that its behaviour could be customized depending on where the sign-out action is triggered from?

**Comment by perarnborg** [comment]
Silent signout at the provider is not yet supported. It has been suggested to add support (#57) but I am not sure when I will have the time to look at that.

**Comment by esune** [comment]
Thanks for the reply, I must have missed that other issue 🙂

**Comment by perarnborg** [comment]
As of v3.8.0 there is a signOutOidcSilent action that can be used for this.

**Comment by esune** [comment]
Thank you! I will update my plugin and code asap and report back.

**Comment by esune** [comment]
I am not sure on whether I am doing something wrong, but it doesn't seem to be working for me: I am using the new `signOutOidcSilent` and the user seem to be signed-out (the authentication info in the header - driven by the current `oidcUser` - disappears), however I can still see the session tokens in the browser storage and when I re-access the resource it detects the previously signed-in user rather than prompting me for authentication again.

**Comment by perarnborg** [comment]
OK, I will have a another look at this!

**Comment by coolhome** [comment]
@esune are you using SilentSignIn as well? You may need to upgrade to v3.9.1 which contains https://github.com/perarnborg/vuex-oidc/pull/122 

Before this patch I noticed it was creating two sessions (silent and redirect) in localstorage, I wonder if this could be part of the problem.

**Comment by esune** [comment]
> 
> 
> @esune are you using SilentSignIn as well? You may need to upgrade to v3.9.1 which contains #122
> 
Yes, I am using `SilentSignIn`. I bumped my version to `3.9.1`, but the user seems to not be completely signed out when dispatching `signOutOidcSilent`. I can see *some* of the cookies set on the URL of the authentication service (in my case Keycloak) are erased (the ones names `..._LEGACY`), however the `AUTH_SESSION_ID` cookie persists so when I switch views in my web application the user appears as authenticated automatically again.

**Comment by perarnborg** [comment]
@esune So signOutOidc works for you (it removes all storage for the session at the provider)?

**Comment by esune** [comment]
I will triple check again as soon as I have a minute, but I believe so - it signs me out and sends me back to the IdP authentication screen

**Comment by perarnborg** [comment]
Ok I have an idea of what can be missing in my implementation. I will provide a patch as soon as I can.

**Comment by perarnborg** [comment]
@esune Could you try again after updating to v3.9.2?

**Comment by perarnborg** [comment]
...sorry, make that v3.9.3

**Comment by esune** [comment]
I tried with the latest v3.9.5: unfortunately it still doesn't completely sign-out the user.

I had a bit more time to poke around and the session cookies may not be the issue (anymore). I noticed that when `signOutOidcSilent` is dispatched, an item named `oidc.user:http://localhost:8180/auth/realms/issuer-kit:issuer-kit-public` remains in the browser's local storage for the application while the same item is cleared when dispatching `signOutOidc`.
